### DISCUSSION TITLE
Enhancing User Experience: Suggestions for Smoother Navigation

### DISCUSSION THREAD

**Discussion by Decryptu** [ebkr/r2modmanPlus#1223] [Ideas]
Hello,

Here are some ideas I had while using the app to make the navigation a bit smoother:

1. **UI Simplification for Sorting and Filtering**:  [Implemented in PR #1226](https://github.com/ebkr/r2modmanPlus/pull/1226) - Replace the "Standard / Reverse" sorting button with a simple caret icon, pointing up for "Standard" and down for "Reverse" to save space and simplify interaction. Optimize "Filter Categories" by replacing the button text with a single font-awesome icon.

2. **Uninstall Button Color Cue**: [Implemented in PR #1224](https://github.com/ebkr/r2modmanPlus/pull/1224) - Change the "Uninstall" button to have a red border-color instead of blue, and a red background color on :hover to more clearly indicate a cautionary action.

3. **Enhance Mod List Visibility**: Propose displaying the download count directly within the mod row in the "Online" mod list in a shortened format, e.g., "1m+" for over a million downloads, "100k+" for over a hundred thousand, etc., for more accessible visibility.

4. **Game Representation**: Suggest adding a small banner representing the current game on the top left corner for better visual representation.
   ![Game Banner](https://github.com/ebkr/r2modmanPlus/assets/90518536/d9de094f-0aa8-434b-a33f-48769f6b0518)

I've started working on some of these ideas and have submitted a pull request that addresses the simplification for sorting and filtering as well as the uninstall button color cue.

If you find the other suggestions relevant, I may try to implement more when I find the time. Anyway, thanks for this awesome app and the opportunity to contribute.

**Comment by ebkr** [comment]
Hey, thanks for the ideas!

> UI Simplification for Sorting and Filtering: https://github.com/ebkr/r2modmanPlus/pull/1226 - Replace the "Standard / Reverse" sorting button with a simple caret icon, pointing up for "Standard" and down for "Reverse" to save space and simplify interaction. Optimize "Filter Categories" by replacing the button text with a single font-awesome icon.

I'm not a huge fan of this idea because relying on icons alone is bad practice. It frees up space but if you aren't familiar with the icons then it leads to people either ignoring the feature or clicking things randomly, neither of which are particularly ideal. Granted the manager isn't very accessible when it comes to keyboard navigation, however for people able to navigate with a mouse, it's quite nice to be able to know what an action does if your eyesight isn't the best.

See https://www.nngroup.com/articles/icon-usability/ for example

> Uninstall Button Color Cue: https://github.com/ebkr/r2modmanPlus/pull/1224 - Change the "Uninstall" button to have a red border-color instead of blue, and a red background color on :hover to more clearly indicate a cautionary action.

I like this idea, however it does however draw attention away from the rest of the content significantly so could probably be nicer with muted colours. I'll comment the same on the PR.

> Enhance Mod List Visibility: Propose displaying the download count directly within the mod row in the "Online" mod list in a shortened format, e.g., "1m+" for over a million downloads, "100k+" for over a hundred thousand, etc., for more accessible visibility.

I agree, it's easy to do and would prevent the tag becoming ridiculously long. It could also have a tooltip with the full count if people wanted to know however equally it's not really important information so I leave that open for general discussion.

> Game Representation: Suggest adding a small banner representing the current game on the top left corner for better visual representation.

This is a funny one because it contradicts your first feature request to save space. I don't think we get any benefit in this over having just the text at the top of the navigation bar. It would just be a change for the sake of looking slightly better, but it does also tie in with the theme on the last two replies that it just takes attention away from the main body of the manager.

**Comment by Decryptu** [comment]
Absolutely, these were just a few thoughts I had while navigating the app. Your feedback clarifies the direction you're heading with the design, and I appreciate the insight. I'll review the comments on the PR for the uninstall button and agree to retract the PR concerning the icons – your points make sense.

Regarding the download count feature, it seems we're on the same page. It was an idea that came to me as I was looking through mods sorted by their last update. I thought it could be beneficial to see the download counts directly, making it easier to spot the more popular mods amidst the newer or less downloaded ones. It seems like a practical addition for users to quickly gauge a mod's popularity at a glance.

**Comment by Decryptu** [comment]
And here just to illustrate the download count:  
![dlcount](https://github.com/ebkr/r2modmanPlus/assets/90518536/9855c60e-98c6-451c-ad76-8222b1555159)

I used this logic, maybe it can be enhanced: 

```vue
// Method to format the download count
formatDownloads(count: number): string {
    if (count >= 1000000) {
        return `${(count / 1000000).toFixed(1)}m+`; // For 1 million or more
    } else if (count >= 1000) {
        return `${Math.floor(count / 1000)}k+`; // For 1,000 to 999,999, always round down
    } else {
        return count.toString(); // For less than 1,000
    }
}
```

and just added this with the other icons in the mod list:

```vue
<span class='card-header-icon'>
    {{ formatDownloads(key.getTotalDownloads()) }} <i class='fas fa-download margin-left--half-width'></i>
</span>
```
